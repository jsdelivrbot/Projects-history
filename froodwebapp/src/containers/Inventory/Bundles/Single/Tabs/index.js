export ConnectedMainTab from './MainTab/MainTab';
export ConnectedItemsTab from './ItemsTab/ItemsTab';
export ConnectedAssemblyTab from './AssemblyTab/AssemblyTab';
export Tabs from './Tabs';
