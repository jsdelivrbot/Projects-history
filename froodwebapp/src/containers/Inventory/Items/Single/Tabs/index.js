export ConnectedMainTab from './MainTab/MainTab';
export ConnectedBinsTab from './BinsTab/BinsTab';
export ConnectedSuppliersTab from './SuppliersTab/SuppliersTab';
export ConnectedUOMTab from './UOMTab/UOMTab';
export StockLifeCycleTab from './StockLifeCycleTab/StockLifeCycleTab';
export ImagesTab from './ImagesTab/ImagesTab';
export Tabs from './Tabs';
