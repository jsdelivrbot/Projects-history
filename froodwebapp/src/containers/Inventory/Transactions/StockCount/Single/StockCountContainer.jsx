/* eslint-disable jsx-a11y/label-has-for */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import ConnectedStockCount from './StockCount/StockCount';

const mapStateToProps = state => ({
  loadingPage: state.priceList.loadingPage,
  priceList: state.priceList.priceList,
  login: state.login.user,
});

export class StockCountContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      stockCountId: this.props.match.params.id,
      isNewStockCount: this.props.match.params.id === 'new'
    };
  }

  componentWillReceiveProps(nextProps) {
    const stockCountId = nextProps.match.params.id;

    this.setState({
      stockCountId,
      isNewStockCount: stockCountId === 'new'
    });
  }


  render() {
    const {
      stockCountId,
      isNewStockCount
    } = this.state;

    return (
      <ConnectedStockCount
        stockCountId={ Number(stockCountId) }
        isNewStockCount={ isNewStockCount }
      />
    );
  }
}

StockCountContainer.propTypes = {
  // router
  match: PropTypes.object.isRequired,
};

export default withRouter(connect(mapStateToProps)(StockCountContainer));
