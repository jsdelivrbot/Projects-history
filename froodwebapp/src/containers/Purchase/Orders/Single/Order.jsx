import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Spin } from 'antd';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import { withRouter } from 'react-router-dom';
import { reduxForm } from 'redux-form';
import {
  purchaseOrderGetRequest,
  purchaseOrderFieldsGetRequest,
} from 'redux-base/actions';
import { form } from 'styles/common.scss';
import { Tabs, ConnectedOrderInfo } from './Sections';
import editableStatuses from './orderHelpers';

const mapStateToProps = state => ({
  loadingPage: state.purchaseOrder.loadingPage,
  initialValues: state.purchaseOrder.initialValues,
  status: state.purchaseOrder.initialValues.status,
});

const mapDispatchToProps = {
  purchaseOrderGetRequest,
  purchaseOrderFieldsGetRequest,
  push
};

const reduxFormConfig = {
  form: 'newPurchaseOrderForm',
  enableReinitialize: true
};

export class Order extends Component {
  constructor(props) {
    super(props);
    const orderId = this.props.match.params.id;
    const isNewOrder = orderId === 'new';

    this.state = {
      orderId,
      isNewOrder
    };
  }

  componentWillMount() {
    if (this.state.isNewOrder) {
      this.props.purchaseOrderFieldsGetRequest();
    } else {
      this.props.purchaseOrderGetRequest({
        params: [{
          id: this.state.orderId
        }]
      });
    }
  }

  componentWillReceiveProps(nextProps) {
    // redirect to all PO screen when create new one
    if ((nextProps.initialValues.id && this.state.isNewOrder) ||
    // when updating existing one to another status
    (nextProps.initialValues.id === this.props.initialValues.id && nextProps.status !== this.props.status)) {
      this.props.push('/purchase/orders');
    }

    const orderId = nextProps.match.params.id;
    const isNewOrder = orderId === 'new';

    this.setState({
      orderId,
      isNewOrder
    });
  }

  render() {
    const {
      isNewOrder,
      orderId
    } = this.state;

    const {
      // trigger
      loadingPage,
      // props
      status,
      // form
      handleSubmit
    } = this.props;

    // disable stuff, when status is not `Draft` or `Not Saved`
    const orderInfoDisabled = !editableStatuses.includes(status);

    return (
      <Spin spinning={ loadingPage }>
        <form className={ form }>
          <ConnectedOrderInfo
            status={ status }
            isNewOrder={ isNewOrder }
            disabled={ orderInfoDisabled }
          />
          <Tabs
            status={ status }
            orderId={ orderId }
            orderInfoDisabled={ orderInfoDisabled }
            handleSubmit={ handleSubmit }
          />
        </form>
      </Spin>
    );
  }
}

Order.propTypes = {
  loadingPage: PropTypes.bool.isRequired,
  // redux-base
  purchaseOrderGetRequest: PropTypes.func.isRequired,
  purchaseOrderFieldsGetRequest: PropTypes.func.isRequired,
  // router
  push: PropTypes.func.isRequired,
  match: PropTypes.object.isRequired,
  // form
  handleSubmit: PropTypes.func.isRequired,
  initialValues: PropTypes.object,
  // data
  status: PropTypes.string.isRequired
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(reduxForm(reduxFormConfig)(Order)));
