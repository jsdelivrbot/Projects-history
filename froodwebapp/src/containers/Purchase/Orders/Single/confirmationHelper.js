import React from 'react';
import { startCase } from 'lodash';
import { Icon } from 'antd';
import diff from 'object-diff';
import {
  confirmItem,
  confirmNewItem,
  confirmDeleteItem
} from './confirmationHelper.scss';

export default (formValues, initialValues) => {
  // order changes
  const orderChanges = diff(formValues, initialValues);

  const modalData = Object.getOwnPropertyNames(orderChanges).map(propName =>
    propName !== 'details' &&
    <p key={ propName } className={ confirmItem }>
      <Icon type="check" />
      { `${startCase(propName)} changed from ` }
      <span>{ `${orderChanges[propName]} ` }</span>
      to
      <span>{ ` ${formValues[propName]}` }</span>
    </p>
  );

  // if added new skuDetails or changed existing skuDetails
  formValues.details.forEach((skuDetail, index) => {
    if (skuDetail.deleted) return;

    const initialSkuDetail = initialValues.details[index];

    if (initialSkuDetail) { // if its not a new item
      const differ = diff(skuDetail, initialValues.details[index]);
      Object.getOwnPropertyNames(differ).map(propName =>
        modalData.push(
          <p key={ propName } className={ confirmItem }>
            <Icon type="check" />
            { `${skuDetail.name} ${startCase(propName)} changed from ` }
            <span>{ `${differ[propName]} ` }</span>
            to
            <span>{ ` ${skuDetail[propName]}` }</span>
          </p>
        )
      );
    } else { // if a new item
      modalData.push(
        <p key={ skuDetail.name } className={ confirmNewItem }>
          <Icon type="check" />
          { `Added ${skuDetail.name} with Qty ` }
          <span>{ `${skuDetail.qty} ` }</span>
          { Number(skuDetail.discount) !== 0 &&
            <span>
              <span>and Discount</span>
              <span>{ ` ${skuDetail.discount}` }</span>
            </span>
          }
        </p>
      );
    }
  });

  // if removed skuDetails
  formValues.details.filter(sku => sku.deleted === true && sku.new !== true)
    .forEach(skuDetail =>
      modalData.push(
        <p key={ skuDetail.name } className={ confirmDeleteItem }>
          <Icon type="check" />
        Removed
          <span>{ ` ${skuDetail.name}` }</span>
        </p>
      )
    );

  return modalData;
};
