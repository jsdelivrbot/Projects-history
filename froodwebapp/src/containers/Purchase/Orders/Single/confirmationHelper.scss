@import '~styles/colors.scss';

.confirmItem {
  color: $main-black;
  margin-bottom: 1rem;
  i {
    margin-right: 1rem;
    color: $main-green;
  }
  span {
    color: $main-red;
  }
  span:last-child {
    color: $main-green;
  }
}

.confirmNewItem {
  @extend .confirmItem;
  span {
    color: $main-green;
  }
  span:last-child {
    span:first-child {
      color: $main-black;
    }
  }
}

.confirmDeleteItem {
  @extend .confirmItem;
  i {
    color: $main-red;
  }
  span {
    color: $main-red;
  }
}
