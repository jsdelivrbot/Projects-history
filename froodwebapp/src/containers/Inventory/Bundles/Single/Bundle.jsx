import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { PageHeader } from 'components';
import { Tabs, ConnectedMainTab } from './Tabs';

export class Bundle extends Component {
  constructor(props) {
    super(props);
    const bundleId = props.match.params.id;
    const isNewBundle = bundleId === 'new';

    this.state = {
      bundleId,
      isNewBundle
    };
  }

  componentWillReceiveProps(nextProps) {
    const bundleId = nextProps.match.params.id;
    const isNewBundle = bundleId === 'new';

    this.setState({
      bundleId,
      isNewBundle
    });
  }

  render() {
    const {
      bundleId,
      isNewBundle
    } = this.state;

    return (
      <div>
        { isNewBundle &&
          <PageHeader
            bigText="Item Bundles"
          />
        }
        { isNewBundle &&
          <ConnectedMainTab
            isNewBundle={ isNewBundle }
          />
        }
        { !isNewBundle &&
          <Tabs
            bundleId={ bundleId }
            isNewBundle={ isNewBundle }
          />
        }
      </div>
    );
  }
}

Bundle.propTypes = {
  // router
  match: PropTypes.object.isRequired,
};

export default withRouter(Bundle);
