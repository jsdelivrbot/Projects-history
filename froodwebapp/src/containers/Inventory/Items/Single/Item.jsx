import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { PageHeader } from 'components';
import { Tabs, ConnectedMainTab } from './Tabs';

export class Item extends Component {
  constructor(props) {
    super(props);
    const itemId = props.match.params.id;
    const isNewSkuItem = itemId === 'new';

    this.state = {
      itemId,
      isNewSkuItem
    };
  }

  componentWillReceiveProps(nextProps) {
    const itemId = nextProps.match.params.id;
    const isNewSkuItem = itemId === 'new';

    this.setState({
      itemId,
      isNewSkuItem
    });
  }

  render() {
    const {
      itemId,
      isNewSkuItem
    } = this.state;

    return (
      <div>
        { isNewSkuItem &&
          <PageHeader
            bigText="Product Details"
            smallText="Enter your product name, type and other details to setup your product"
          />
        }
        { isNewSkuItem &&
          <ConnectedMainTab
            isNewSkuItem={ isNewSkuItem }
          />
        }
        { !isNewSkuItem &&
          <Tabs
            itemId={ itemId }
            isNewSkuItem={ isNewSkuItem }
          />
        }
      </div>
    );
  }
}

Item.propTypes = {
  match: PropTypes.object.isRequired,
};

export default withRouter(Item);
