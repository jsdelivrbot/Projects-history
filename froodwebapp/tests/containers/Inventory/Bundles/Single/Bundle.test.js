import { Bundle } from 'containers/Inventory/Bundles/Single/Bundle';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('Bundle', () => {
  const props = {
    match: {
      params: {
        id: 'some id'
      }
    }
  };
  it('renders the Bundle component', () => {
    const component = shallow(<Bundle { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of Bundle', () => {
    const component = shallow(<Bundle { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });
});
