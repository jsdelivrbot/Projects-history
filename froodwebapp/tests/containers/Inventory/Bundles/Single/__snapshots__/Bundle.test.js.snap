// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bundle capturing Snapshot of Bundle 1`] = `
<div>
  <withRouter(Connect(ItemBundlesTabs))
    bundleId="some id"
    isNewBundle={false}
  />
</div>
`;
